packages/protocolbeat/src/apps/discovery/hooks/useTemplateModel.ts

```diff
@@ -0,0 +1,98 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { writeTemplateFile } from '../../../api/api'
+import { toggleInList } from '../../../utils/toggleInList'
+import { ContractConfigModel } from '../components/ConfigModel'
+
+type Props = {
+  project: string
+  templateId?: string
+  files: {
+    template: string
+    shapes?: string
+    criteria?: string
+  }
+}
+
+export function useTemplateModel({ templateId, files }: Props) {
+  const queryClient = useQueryClient()
+  const [templateModel, setTemplateModel] = useState(
+    ContractConfigModel.fromRawJsonc(files.template),
+  )
+
+  useEffect(() => {
+    setTemplateModel(ContractConfigModel.fromRawJsonc(files.template))
+  }, [files.template])
+
+  const toggleIgnoreMethods = (fieldName: string) => {
+    const current = templateModel.ignoreMethods ?? []
+    const updated = toggleInList(fieldName, current)
+    const newModel = templateModel.setIgnoreMethods(updated)
+    setTemplateModel(newModel)
+    saveMutation.mutate(newModel.toString())
+  }
+
+  const toggleIgnoreRelatives = (fieldName: string) => {
+    const current = templateModel.ignoreRelatives ?? []
+    const updated = toggleInList(fieldName, current)
+    const newModel = templateModel.setIgnoreRelatives(updated)
+    setTemplateModel(newModel)
+    saveMutation.mutate(newModel.toString())
+  }
+
+  const toggleIgnoreInWatchMode = (fieldName: string) => {
+    const current = templateModel.ignoreInWatchMode ?? []
+    const updated = toggleInList(fieldName, current)
+    const newModel = templateModel.setIgnoreInWatchMode(updated)
+    setTemplateModel(newModel)
+    saveMutation.mutate(newModel.toString())
+  }
+
+  const saveMutation = useMutation({
+    mutationFn: async (content?: string) => {
+      if (!templateId) {
+        return
+      }
+      await writeTemplateFile(templateId, content ? content : templateString)
+    },
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({
+        queryKey: ['template', templateId],
```

Comment:
Is this correct? In useCurrentTemplate.ts it's `['projects', project, 'template', templateId, selectedAddress]`